runway/module/staticsite.py

```diff
@@ -13,126 +12,195 @@
                  ]},
                 output_stream,
                 default_flow_style=False
             )
-        site_stack_variables = {
-            'AcmCertificateArn': '${default staticsite_acmcert_arn::undefined}',
-            'Aliases': '${default staticsite_aliases::undefined}',
-            'DisableCloudFront': '${default staticsite_cf_disable::undefined}',
-            'RewriteDirectoryIndex': '${default staticsite_rewrite_directory_index::undefined}',  # noqa pylint: disable=line-too-long
-            'WAFWebACL': '${default staticsite_web_acl::undefined}'
-        }
 
-        parameters = self.options['parameters']
+    def _create_staticsite_yaml(self, module_dir):
+        # Default parameter name matches build_staticsite hook
+        hash_param = self.user_options.get('source_hashing', {}).get(
+            'parameter',
+            "${namespace}-%s-hash" % self.name
+        )
 
-        if parameters.get('staticsite_acmcert_ssm_param'):
-            site_stack_variables['AcmCertificateArn'] = '${ssmstore ${staticsite_acmcert_ssm_param}}'  # noqa pylint: disable=line-too-long
+        build_staticsite_args = self.options.copy() or {}
+        build_staticsite_args['artifact_bucket_rxref_lookup'] = "%s-dependencies::ArtifactsBucketName" % self.name  # noqa pylint: disable=line-too-long
+        build_staticsite_args['options']['namespace'] = '${namespace}'
+        build_staticsite_args['options']['name'] = self.name
+        build_staticsite_args['options']['path'] = os.path.join(
+            os.path.realpath(self.context.env_root),
+            self.path
+        )
 
-        if parameters.get('staticsite_enable_cf_logging', True):
-            site_stack_variables['LogBucketName'] = "${rxref %s-dependencies::AWSLogBucketName}" % name  # noqa pylint: disable=line-too-long
+        site_stack_variables = self._get_site_stack_variables()
+
+        class_path = 'staticsite.StaticSite'
+        bucket_name = ''
+
+        pre_build = [{'path': 'runway.hooks.staticsite.build_staticsite.build',
+                      'required': True,
+                      'data_key': 'staticsite',
+                      'args': build_staticsite_args}]
+
+        post_build = [{'path': 'runway.hooks.staticsite.upload_staticsite.sync',
+                       'required': True,
+                       'args': {
+                           'bucket_output_lookup': '%s::BucketName' % self.name,
+                           'website_url': '%s::BucketWebsiteURL' % self.name,
+                           'cf_disabled': site_stack_variables['DisableCloudFront'],
+                           'distributionid_output_lookup': '%s::CFDistributionId' % (self.name),
+                           'distributiondomain_output_lookup': '%s::CFDistributionDomainName' % self.name}}]  # noqa pylint: disable=line-too-long
+
+        if self.parameters.get('staticsite_auth_at_edge', False):
+            bucket_name = '%s-aae-lambdas' % self.parameters.get('namespace')
```

Comment:
@voodooGQ we had tested using the bucket from the dependencies stack before (by using an rxref lookup on the stack), but the lookup didn't work correctly (left literal lookup text in place instead of performing the lookup).

Can you try swapping that in again and checking with Kyle if it still isn't working?

Reply:
@ITProKyle 
![image](https://user-images.githubusercontent.com/1512028/75785036-bc73c480-5d28-11ea-80d8-38ee8a2330b1.png)
![image](https://user-images.githubusercontent.com/1512028/75785049-c3023c00-5d28-11ea-8eb6-975d705bc484.png)

Reply:
Appears to be getting parsed prior to the swap in.

Reply:
@voodooGQ - the error you are getting is due to the use of `bucket_name` here - https://github.com/onicagroup/runway/blob/44207fc4f3833d72dbb93c37c54de8b085a852ae/runway/module/staticsite.py#L192

CFNgin lookups are not resolved in `cfngin_bucket`/`stacker_bucket` - only stack variables and hook args

Reply:
@ITProKyle Does that imply there's no way for us to use the ArtifactBucket created in the dependencies step as the main stacker_bucket for the primary stack? We're trying to see if we can share the resource. Do you know an alternative way to accomplish that?

Reply:
are the CFN templates small enough to be used through the API without placing them in S3? If thats the case and you only the bucket to store lamda src code/passed to your stacks you can use `cfngin_bucket: ''` to precent CFN template caching and provide `bucket: ${rxref ...}` as n arg to the aws_lambda hook and stacks as needed.

Reply:
I'll leave it up to @troyready on how he wants to approach it.

Reply:
Probably easiest to leave `cfngin_bucket: ''` (don't need to cache the templates on s3), and just handle getting the bucket name in the function doing the staging of the lambda folders.